### PR TITLE
[test] include oss tar in packaging tests

### PR DESCRIPTION
**Pull request by andyb-elastic**
Add the oss tar distribution to the packaging test plugin. Test the oss
tar distribution in the core packaging tests, and the non-oss tar
distribution in the x-pack packaging tests.

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]
This makes sense to me. I expect @rjernst will also want to take a look at it.

File: qa/vagrant/src/test/resources/packaging/utils/tar.bash [hunk 1]
```diff
@@ -79,6 +81,8 @@ export_elasticsearch_paths() {
     export ESSCRIPTS="$ESCONFIG/scripts"
     export ESDATA="$ESHOME/data"
     export ESLOG="$ESHOME/logs"
+
+    export ARCHIVE_NAME=${ARCHIVE_NAME:-"elasticsearch-oss"}
```
Comment (rjernst) [inline qa/vagrant/src/test/resources/packaging/utils/tar.bash]: Can we just reuse PACKAGE_NAME?

Comment (andyb-elastic) [reply qa/vagrant/src/test/resources/packaging/utils/tar.bash]: Yeah I'll do that

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**